clientGQL_test.go

```diff
@@ -71,33 +71,33 @@ func FixtureQueryValidation(t *testing.T, fixture string) autopilot.RequestValid
 }
 
 func ATestClient(t *testing.T, endpoint string) *ol.Client {
-	return ol.NewGQLClient(ol.SetAPIToken("x"), ol.SetURL(autopilot.RegisterEndpoint(fmt.Sprintf("/LOCAL_TESTING/%s", endpoint),
+	return ol.NewGQLClient(ol.SetAPIToken("x"), ol.SetMaxRetries(0), ol.SetURL(autopilot.RegisterEndpoint(fmt.Sprintf("/LOCAL_TESTING/%s", endpoint),
```

Comment:
I figured this would have been a compilation error without the maxRetries?

Reply:
Hmm not sure what you mean.  What i'm doing his is removing the default amount of retries on the test client so that It only tries 1 time instead of 5 - which hugely slows down when tests are fail.  This makes them snappy again when you have a failing test.

Reply:
This is probably a go method signature thing,  you've added a new positional parameter to `NewGQLClient` call based on my reading here.

Reply:
Ahh - yes this is the "spread" operator on parameters - AKA [Variadic Function Parameters](https://yourbasic.org/golang/three-dots-ellipsis/)

The signature of NewGQLClient use `...` on the `Option` type which is to take in any N arguments such that variable `options` is effectively of type `[]Option`.
```
func NewGQLClient(options ...Option) *Client {
```

This is also using the pattern dubbed [Functional Options](https://golang.cafe/blog/golang-functional-options-pattern.html#:~:text=The%20Go%20(Golang)%20Functiona%20Options,and%20initialisation%20of%20your%20struct.)

and then latter in the function we do this to apply all the "Functional Options" to the client before returning it.

```
        settings := &ClientSettings{
	for _, opt := range options {
		opt(settings)
	}
```

Reply:
Ah thanks for the clarification!